ArjunaJTS/jtax/tests/classes/com/hp/mwtests/ts/jta/jts/xa/XAResourceRecoveryTest.java

```diff
@@ -101,8 +103,10 @@ public void testRMFAILOutOfRollback() throws Exception {
         recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryModules(Arrays.asList(new RecoveryModule[]{xaRecoveryModule}));
 
         assertNotNull("Could not find recovery state before recovery fails", StoreManager.getParticipantStore().read_committed(uid, XAResourceRecord.typeName()));
-        RecoveryManager.manager().scan();
+        RecoveryManager.manager().scan(); // First pass should be XAER_RMFAIL
         assertNotNull("Could not find recovery state after recovery fails", StoreManager.getParticipantStore().read_committed(uid, XAResourceRecord.typeName()));
+        RecoveryManager.manager().scan(); // Second pass should be XAER_NOTA
+        assertNull("Found recovery state after receiving XAER_NOTA", StoreManager.getParticipantStore().read_committed(uid, XAResourceRecord.typeName()));
```

Comment:
ok so this is validating that we retry after the first XAER_RMFAIL.

Reply:
It's a little more than that. Basically the test is simulating first getting the RMFAIL and then the NOTA. So +1 it is validating we retry, but slightly more than that it is validating that NOTA will (expectedly) clean up the state on disk.